app/src/main/java/com/github/geohunt/app/ui/screens/home/ChallengeFeed.kt

```diff
@@ -0,0 +1,106 @@
+package com.github.geohunt.app.ui.screens.home
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.Button
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.github.geohunt.app.model.Challenge
+import com.github.geohunt.app.model.User
+import com.github.geohunt.app.ui.components.appbar.HomeScreenFeed
+import com.github.geohunt.app.ui.components.cards.ChallengeCard
+
+@Composable
+fun ChallengeFeed(
+    chosenFeed: HomeScreenFeed,
+    onUserClick: (User) -> Unit,
+    onOpenMap: (Challenge) -> Unit,
+    onOpenChallenge: (Challenge) -> Unit,
+    onClaim: (Challenge) -> Unit,
+    onOpenExplore: () -> Unit,
+    viewModel: HomeFeedViewModel = viewModel(factory = HomeFeedViewModel.Factory)
+) {
+    val feed = viewModel.getFeed(chosenFeed)
+    val challenges = feed.challenges.collectAsStateWithLifecycle()
+
+    when {
+        challenges.value == null -> Box(modifier = Modifier.fillMaxSize()) {
+            CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
+        }
+
+        challenges.value!!.isEmpty() -> Column(modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp),
+            verticalArrangement = Arrangement.Center
+        ) {
+            Text(
+                "This place is a bit empty.",
```

Comment:
String resources? xd

Reply:
XD

Reply:
Had to write code fast, changing it ASAP 👍

Reply:
tkt mon cœur